sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -44,20 +44,34 @@ namespace Azure { namespace Core {
   class Context final {
   public:
     /**
-     * @brief A context key.
+     * @brief A key used to store and retrieve data in a Context object.
      *
      */
     class Key final {
       Key const* m_uniqueAddress;
 
     public:
+      /**
+       * @brief Constructs a Key
+       *
+       */
       Key() : m_uniqueAddress(this) {}
 
+      /**
+       * @brief Compare with \p other #Azure::Core::Context::Key for equality.
+       * @param other Other #Azure::Core::Context::Key to compare with.
+       * @return `true` if #Azure::Core::Context::Key instances are equal; otherwise, `false`.
+       */
       bool operator==(Key const& other) const
       {
         return this->m_uniqueAddress == other.m_uniqueAddress;
       }
 
+      /**
+       * @brief Compare with \p other #Azure::Core::Context::Key for equality.
```

Comment:
1. Compares...
2. Remove self-reference

Reply:
If you want to use the name of the class, such as "Compare with another `Key`", backtick the `Key`.